### ISSUE TITLE
babel language didn't change after change the player faction in game

### ISSUE THREAD

**Issue by eststar** [IDI-Systems/acre2#435]
**Arma 3 Version:** 1.76.143187 (stable )
**CBA Version:** `3.4.1.170912` (stable)
**ACRE2 Version:** `2.5.0.968` (stable)

**Mods:**
```
- CBA_A3
- acre2
-ace
-Ares Mod - Achilles Expansion

```

**Description:**
- When I change the faction of player, the babel language selection didn't changed.
I mean like I used babel setting "per side /w common"
and I added script at intitplayerlocal.sqf like below

[[west, "English"],[east, "Russian"],[independent, "Arabic"],[civilian,"Arabic", "English", "Russian"]] call acre_api_fnc_setupMission;
    ["en", "English"] call acre_api_fnc_babelAddLanguageType;
    ["ru", "Russian"] call acre_api_fnc_babelAddLanguageType;
	["ab", "Arabic"] call acre_api_fnc_babelAddLanguageType;

	if (hasInterface) then {
        switch ( side (group player) ) do {
            case west: {
                ["en"] call acre_api_fnc_babelSetSpokenLanguages;
            };
            case civilian: {
                ["ab","en","ru"] call acre_api_fnc_babelSetSpokenLanguages;
            };
            case east: {
                ["ru"] call acre_api_fnc_babelSetSpokenLanguages;
            };
            case independent: {
                ["ab"] call acre_api_fnc_babelSetSpokenLanguages;
            };
        };
    };

And this script works well without change the side of player in game(zeus)

I think it is not error but intended, I hope in game zeus-change-side can effect to acre babel system. 
Or maybe make some default babel system module or script something.
Thanks for read.

**Steps to reproduce:**
- Add the steps needed to reproduce the issue.

**Where did the issue occur?**
- Editor (Multiplayer) 

**Placed Modules:**
- basic mission setup(acre2), ace medical modules, ace pointing, ace rearm, ace repair, and ace some more ace modules.

**RPT log file:**
-

**Comment by TheMagnetar** [comment]
When changing the side through zeus, initplayerlocal is not executed, unless I am mistaken.

**Comment by eststar** [comment]
I changed addon list. And then any way to check that script for check the side in game?

**Comment by TheMagnetar** [comment]
You can always use the debug console. You can activate it in Eden Editor.

**Comment by jonpas** [comment]
Is there an event handler for side change?

**Comment by PabstMirror** [comment]
We use something like this
```
["unit", {
    params ["_player"];
    switch ((getNumber (configFile >> "CfgVehicles" >> (typeOf _player) >> "side"))) do {
        case 1: { ["en"] call acre_api_fnc_babelSetSpokenLanguages; };
        case 0: { ["ru"] call acre_api_fnc_babelSetSpokenLanguages; };
        case 2: { ["ab"] call acre_api_fnc_babelSetSpokenLanguages; };
        default {  ["ab","en","ru"] call acre_api_fnc_babelSetSpokenLanguages; };
    };
}, true] call CBA_fnc_addPlayerEventHandler;
```

**Comment by jonpas** [comment]
Since changing faction often means you likely want different languages, I don't think it's even something we want to do in ACRE2 itself.

**Comment by TheMagnetar** [comment]
I would leave it out of ACRE2 as well.

**Comment by eststar** [comment]
Ok I understand... I will find another way. Thanks anyway

**Comment by jonpas** [comment]
@eststar you were given a solution in https://github.com/IDI-Systems/acre2/issues/435#issuecomment-341909236

Maybe we could add that as a setting? Disabled by default.

Reopening for discussion.

**Comment by eststar** [comment]
Actually I am such a noob for these arma 3 scripts, and somehow my previous babel script is not working... So I am confused now....  Where should I put that #435 script, init? description? initplayerlocal?
And I hope there is some module or option for babel /w common. It is hard to add script in mission for noob scripter......

**Comment by eststar** [comment]
@PabstMirror  are you use that  script for babel setting? or for event handler for babel setting? And is that script should be inside initplayerlocal or else where?

**Comment by eststar** [comment]
@jonpas I was tested that first script with bunch of addons previously, and some how I changed mods to that first one. And I think that works, but it didn't. I am now finding what is make that script works. So... maybe take some time. And anyway, is that babel system works without any addons? like only with cba_a3, and acre2? If not, let me know.. thanks.

**Comment by eststar** [comment]
eventually I found another issue but I don't know what is the problem

in these mods below
- CBA_A3
- acre2
- ace
- MCC Sandbox 4

this script is working when added in intitplayerlocal.sqf.
//
[[west, "English"],[east, "Russian"],[independent, "Arabic"],[civilian,"Arabic", "English", "Russian"]] call acre_api_fnc_setupMission;
["en", "English"] call acre_api_fnc_babelAddLanguageType;
["ru", "Russian"] call acre_api_fnc_babelAddLanguageType;
["ab", "Arabic"] call acre_api_fnc_babelAddLanguageType;

if (hasInterface) then {
    switch ( side (group player) ) do {
        case west: {
            ["en"] call acre_api_fnc_babelSetSpokenLanguages;
        };
        case civilian: {
            ["ab","en","ru"] call acre_api_fnc_babelSetSpokenLanguages;
        };
        case east: {
            ["ru"] call acre_api_fnc_babelSetSpokenLanguages;
        };
        case independent: {
            ["ab"] call acre_api_fnc_babelSetSpokenLanguages;
        };
    };
};
//
But when I put out MCC, babel language didn't work. 
I just use the basic acre module and zeus.
Here is my mission file. https://drive.google.com/open?id=1Hv5APUFlxtY5JwM19K8CXbhblVYhK6U8


And still I want to know which way I can change available babel language with changing faction in game by zeus.

**Comment by TheMagnetar** [comment]
You can use the command "faction" in order to retrieve the unit's faction. The script will look really similar.

**Comment by MaHuJa** [comment]
The current solution/workaround is that you somehow need to manually need to make `["ru"] call acre_api_fnc_babelSetSpokenLanguages;` run for the particular client whose side you just reassigned.
This could be done with something like `if (name player == "thisguy") then { ["ru"] call acre_api_fnc_babelSetSpokenLanguages; };` run globally, which is easier to do with most tools.

It's not hard to think of QOL improvements to that.

**Comment by jonpas** [comment]
@MaHuJa Pull requests are welcome.

**Comment by MaHuJa** [comment]
Maybe next year - for now I'm spending all* acre dev time on 264.  (*not that this says much)

**Comment by jonpas** [comment]
We decided not to implement anything for that in ACRE itself, as there is no way we can satisfy all use cases. After-all the default would stay as it is, when you change faction, you change it due to "allegiance" (who shoots who in Arma pretty much), you don't instantly learn other side's languages and forget the ones you just knew. Additionally custom languages can be defined, so that's another thing that would have to be handled.

@PabstMirror 's solution is by far the best option to go for here, we decided to put it on the wiki (with @PabstMirror 's permission of course) and that should satisfy anyone wanting that functionality.
